docs/contributor/arch/015-transform-filter-support-for-telemetry-pipelines.md

```diff
@@ -0,0 +1,88 @@
+# 15. Transform and Filter support for OpenTelemetry based Pipelines
+
+Date: 2024-10-10
+
+## Status
+
+Proposed
+
+## Context
+
+In the default setup of Metric and Trace Pipelines, users currently lack the ability to filter or transform/enrich data before it is sent to the backend. To provide more flexibility, users should be able to filter data based on specific conditions and apply transformations or enrichment to the data. This solution must ensure a consistent approach across all OpenTelemetry (OTel) pipelines.
+
+## Decision
+
+We will implement a consolidated solution in the OpenTelemetry Collector (OTel Collector) using a single Filter and Transform Processor. This processor will leverage the OpenTelemetry Transformation and Transport Language (OTTL) to handle both filtering and transformation tasks. The processor will be configurable to meet user requirements, and only a subset of OTTL functions will be supported, focusing on the most common and impactful use cases.
```

Comment:
```suggestion
We will implement a consolidated solution in the OpenTelemetry Collector (OTel Collector) using a single Filter and Transform Processor. This processor will use the OpenTelemetry Transformation and Transport Language (OTTL) to handle both filtering and transformation tasks. Users will be able to configure the processor as they need; and only a subset of OTTL functions will be supported, focusing on the most common and impactful use cases.
```

Why do we only support a subset? I believe the reason is mentioned in line 70; but it might make more sense to put it here (as well), so it justifies the `## Decision`.